packages/core/src/components/select/select.tsx

```diff
@@ -48,10 +50,80 @@ export class AtomSelect {
   @Event() atomDismiss!: EventEmitter<void>
   @Event() atomFocus!: EventEmitter<void>
 
+  @Method()
+  setTag() {
+    /**
+     * This method was necessary because the `ion-selection-option` loop does not allow customizations or custom components.
+     * So, to be able to add custom elements such as a tag or a badge inside an option of the `select` field, when the select
+     * is opened, the `onBlur` event triggers this method that performs a search for all `ion-item` elements (which is the
+     * final element rendered to list options) and filters the ones that need to be changed.
+     */
+
+    const ionItemElements = document.querySelectorAll('ion-item') || []
+
+    ionItemElements?.forEach((itemElement) => {
+      const optionText = itemElement.textContent?.trim()
+      const optionWithTag = this.optionsWithTag[optionText]
+
+      if (!optionWithTag) return
+
+      const { color, label } = optionWithTag.tag
+
+      const optionElement =
+        this.getElementByTag(itemElement, 'ion-radio') ||
+        this.getElementByTag(itemElement, 'ion-checkbox')
```

Comment:
<!-- Praises highlight something positive. Try to leave at least one of these comments per review. Do not leave false praise (which can actually be damaging). Do look for something to sincerely praise.  -->
**praise:** ‏Nice that you used `itemElement` to optimize the search.